config/mysqlpersister/mysqlpersister.go

```diff
@@ -0,0 +1,215 @@
+package mysqlpersister
+
+import (
+	"database/sql"
+	"errors"
+	"sort"
+	"sync"
+	"time"
+
+	sq "github.com/Masterminds/squirrel"
+	_ "github.com/go-sql-driver/mysql"
+	"github.com/golang/protobuf/proto"
+
+	"github.com/square/quotaservice/config"
+	"github.com/square/quotaservice/logging"
+	qsc "github.com/square/quotaservice/protos/config"
+)
+
+type MysqlPersister struct {
+	latestVersion int
+	db            *sql.DB
+	m             *sync.RWMutex
+
+	watcher         chan struct{}
+	shutdown        chan struct{}
+	fetcherShutdown chan struct{}
+
+	configs map[int]*qsc.ServiceConfig
+}
+
+type configRow struct {
+	Version int    `db:"Version"`
+	Config  string `db:"Config"`
+}
+
+type Connector interface {
```

Comment:
Defining the interface here right before accepting as a parameter in `New` is a perfect, idiomatic use. So beautiful; I love it.